tests/python/test_ranking.py

```diff
@@ -0,0 +1,39 @@
+import numpy as np
+from scipy.sparse import csr_matrix
+import xgboost
+
+def test_ranking_with_weighted_data():
+    Xrow = np.array([1, 2, 6, 8, 11, 14, 16, 17])
+    Xcol = np.array([0, 0, 1, 1,  2,  2,  3,  3])
```

Comment:
Note that under this example, the current implementation of AUCPR fails with error
```
AUC-PR: AUC > 1.0
```